ldurniat/berry.lua

```diff
@@ -447,484 +468,553 @@ end
 -- @param tilesetsDirectory The path to tilesets.
 -- @return The newly created map.
 ------------------------------------------------------------------------------------------------
-function M.new( filename, tilesetsDirectory )
-
-	tilesetsDirectory = tilesetsDirectory and tilesetsDirectory .. '/' or ''
 
+function Map:new( filename, tilesetsDirectory )
 	-- Read map file
     local data = json.decodeFile( system.pathForFile( filename, system.ResourceDirectory ) )
 
-    -- Create group containg all objects including another groups
-    local map  = display.newGroup()
+	local map = setupDisplayGroup( self )
 
-    -- Store the flipped states
-    local flip, tilesets = {}, data.tilesets
-    local tileLayer, objectLayer, layer, tileset, nextTileset, objects, object, image
+	map.tilesetsDirectory = (tilesetsDirectory and tilesetsDirectory .. '/') or ''
 
     -- Purpose of computation here is simplification of code
-    for i=1, #tilesets do 
+    for i, tileset in ipairs(data.tilesets) do
 
     	-- The tilesets are sorted always in ascending order by their firstgid
-    	tileset                = tilesets[i]
-    	nextTileset            = tilesets[i + 1]
+    	local nextTileset      = data.tilesets[i + 1]
     	tileset.lastgid        = findLastGID( tileset, nextTileset ) 
     	tileset.sequenceData   = buildSequences( tileset )
 
     end
-	
-	for i=1, #data.layers do
 
-		layer = data.layers[i]
+    -- Apply properties from data
+    map.tilesets = data.tilesets
+    map.orientation = data.orientation			
+    map.staggeraxis = data.staggeraxis
+    map.staggerindex = data.staggerindex
+    map.tilewidth = data.tilewidth
+    map.tileheight = data.tileheight
 
-		-- Make sure we have a properties table
-		layer.properties = layer.properties or {}
+	
+	for _, info in ipairs(data.layers) do
+
+		local layer = display.newGroup() 
+
+		-- Apply properties from info
+	    layer.name       = info.name
+	    layer.type       = info.type
+	    layer.alpha      = info.opacity
+	    layer.isVisible  = info.visible
+		layer.offset_x   = info.offsetx or 0
```

Comment:
In my opinion you should just apply offset to layer instead for each  individual object. It is simpler solution.
```
layer.x   =  info.offsetx or 0
```
The same apply for y coordinate.
However, this solution requires some change in  `centerAnchor` function. You need replace 
```
object.x = actualCenterX
object.y = actualCenterY
```
with `object.x, object.y = object.parent:contentToLocal( actualCenterX, actualCenterY )`.

Reply:
Oh wow.  Hadn’t considered this before.  This would definitely be a good idea to implement.  I completely forgot that you can move the entire position of a display group including all the objects inside it.

Reply:
Changes have been made.  Will upload commit tomorrow.

Reply:
Commit 26b9778 addresses this.